docs/src/apalache/tuning.md

```diff
@@ -66,16 +70,23 @@ the actions in the TLA+ spec. To find the numbers, run Apalache with
 ### Invariant filter
 
 `search.invariantFilter=<regex>`. Check the invariant only at the steps that
-satisfy the regular expression.
+satisfy the regular expression. The regular expression should recognize words
+over of the form 's->ki', where `s` is a step number, `k` is an invariant kind
+(["state" or "action"][invariants]) and `i` is a invariant number.
+
+For instance, `search.invariantFilter=10->.*|15->state0|20->action1` tells the
+model checker to check
 
-For instance, `search.invariantFilter=10|15|20` tells the model checker to
-check the invariant only *after* exactly 10, 15, or 20 step were made. Step 0
-corresponds to the initialization with ``Init``, step 1 is the first step
-with ``Next``, etc.
+* all invariants only *after* exactly 10 steps have been made,
+* the *first* state invariant only after exactly 15 steps, and
+* the *second* action invariant after exactly 20 steps.
```

Comment:
Ah, I can see how this could be confusing!
The first clause does not apply from step 10 onwards, but only *at* step 10 (hence, "*exactly* 10 steps").
Do you think it would help to reword this? Or maybe it's enough to change the emphasis from "after" to "exactly"?

Reply:
Yes, that does clear things up for me! I would suggestion perhaps something like:

```suggestion
* *all* invariants (because of the wildcard match) at the 10th step,
* the *first* state invariant (`state0`) at the 15th step, and
* the *second* action invariant (`action1`) at the 20th step.
```

Reply:
I think we're getting close :)
Still one issue though: saying "at the Xth step" here is inaccurate, as it might refer to both the states before and after taking the step. Technically, we're actually checking it at all `(step+1)`th (symbolic) states. How about this?

```suggestion
* *all* invariants (because of the wildcard match) at states reached after the 10th step,
* the *first* state invariant (`state0`) at states reached after the 15th step, and
* the *second* action invariant (`action1`) at states reached after the 20th step.
```

Reply:
My issue here is that in this context I take "after" to mean "anytime from this point forward". I guess its because while you mean the plural "states" to range breadth wise over the immediate successor steps, I think the natural reading is to read it as both breadth- and depth-wise, ranging over all states that could follow.

Maybe using a phrase like "immediate successor states" would help?

Reply:
I guess that's why Igor's original wording was "after exactly 10 steps", which to me communicates that it refers only to 10, not 9 or 11.
But I think you resolve the order of "after" and "exactly" differently than I do?

Reply:
True “after exactly 5 minutes” means to me “any time > t+5m”. Could be i am unusually fixated on an indeterminate range for “after” :)

I don’t want to get you hung up here, especially not if it’s consistent with the other docs. Maybe just merge it as is and we can revisit if other users have trouble?